src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -876,15 +876,16 @@ public Task<bool> HasSuccessfullyLoadedAsync(SolutionState solution, Cancellatio
             private AsyncLazy<VersionStamp> _lazyDependentVersion;
             private AsyncLazy<VersionStamp> _lazyDependentSemanticVersion;
 
-            public async Task<VersionStamp> GetDependentVersionAsync(SolutionState solution, CancellationToken cancellationToken)
+            public Task<VersionStamp> GetDependentVersionAsync(SolutionState solution, CancellationToken cancellationToken)
             {
                 if (_lazyDependentVersion == null)
                 {
+                    var tmp = solution; // temp. local to avoid a closure allocation for the fast path
```

Comment:
:memo: I'm planning to propose an analyzer to enforce this as a constraint so that should help prevent regressions.